src/vfs.c

```diff
@@ -2656,7 +2663,8 @@ static int vfsWalRestore(struct vfsWal *w,
 		}
 		frames[i] = frame;
 
-		p = &data[VFS__WAL_HEADER_SIZE + i * vfsFrameSize(page_size)];
+		offset = (size_t)VFS__WAL_HEADER_SIZE + ((size_t)i * (size_t)vfsFrameSize(page_size));
+		p = &data[offset];
```

Comment:
This indexes into a data buffer of maximum size `SIZE_MAX` and we already check here https://github.com/canonical/dqlite/blob/5d00a628786a4c078e66e4c8dec9e8232e440d9f/src/fsm.c#L464 if the buffer is smaller than `SIZE_MAX`, so this should not overflow. (same reason as above)